tests/foreman/ui_airgun/test_location.py

```diff
@@ -27,3 +37,351 @@ def test_positive_create(session):
+        loc_values = session.location.read(loc.name)
+        assert len(loc_values['users']['resources']['assigned']) == 0
+        assert user.login in loc_values['users']['resources']['unassigned']
+
+
+@run_only_on('sat')
+@skip_if_bug_open('bugzilla', 1321543)
+@tier2
+def test_positive_update_with_all_users(session):
+    """Create location and add user to it. Check and uncheck 'all users'
+    setting. Verify that user is assigned to location and vice versa
+    location is assigned to user
+
+    :id: 17f85968-4aa6-4e2e-82d9-b01bc17031e7
+
+    :customerscenario: true
+
+    :expectedresults: Location and user entities assigned to each other
+
+    :BZ: 1479736
+
+    :CaseLevel: Integration
+    """
+    user = entities.User().create()
+    loc = entities.Location().create()
+    with session:
+        session.organization.select(org_name='Any Organization')
+        session.location.select(loc_name=loc.name)
+        session.location.update(
+            loc.name, {'users.resources.assigned': [user.login]})
+        loc_values = session.location.read(loc.name)
+        user_values = session.user.read(user.login)
+        assert loc_values['users']['resources']['assigned'][0] == user.login
+        assert user_values['locations']['resources']['assigned'][0] == loc.name
+        session.location.update(loc.name, {'users.all_users': True})
+        user_values = session.user.read(user.login)
+        assert loc.name in user_values['locations']['resources']['assigned']
+        session.location.update(loc.name, {'users.all_users': False})
+        user_values = session.user.read(user.login)
+        assert loc.name in user_values['locations']['resources']['unassigned']
+
+
+@run_only_on('sat')
+@skip_if_bug_open('bugzilla', 1321543)
+@tier2
+def test_positive_update_with_all_users_setting_only(session):
+    """Create location and do not add user to it. Check and uncheck
+    'all users' setting. Verify that for both operation expected location
+    is assigned to user
+
+    :id: 6596962b-8fd0-4a82-bf54-fa6a31147311
+
+    :expectedresults: Location entity is assigned to user after checkbox
+        was enabled and then disabled afterwards
+
+    :BZ: 1321543
+
+    :CaseLevel: Integration
+    """
+    user = entities.User().create()
+    loc = entities.Location().create()
+    with session:
+        session.organization.select(org_name='Any Organization')
+        session.location.select(loc_name=loc.name)
+        session.location.update(loc.name, {'users.all_users': True})
+        user_values = session.user.read(user.login)
+        assert loc.name in user_values['locations']['resources']['assigned']
+        session.location.update(loc.name, {'users.all_users': False})
+        user_values = session.user.read(user.login)
+        assert loc.name in user_values['locations']['resources']['unassigned']
+
+
+@run_only_on('sat')
+@tier2
+def test_positive_update_hostgroup(session):
```

Comment:
not sure why did you decide to test functionality in reverse order as it doesn't actually test remove functionality

Reply:
That means
```
         session.location.update(
             loc.name,
             {'host_groups.all_hostgroups': False,
              'host_groups.resources.unassigned': [hostgroup.name]}
         )
 
         loc_values = session.location.read(loc.name)
         assert loc_values[
             'host_groups']['resources']['unassigned'][0] == hostgroup.name
```
```
         assert len(loc_values[
             'host_groups']['resources']['unassigned']) == 1
```
?